de1plus/plugins/web_api/plugin.tcl

```diff
@@ -34,84 +33,112 @@ namespace eval ::plugins::${plugin_name} {
             after 2000 [list info_page [translate "Please change the Webserver auth key!"] [translate "Ok"]]
             return
         }
-        # Define handlers
-        ::wibble::handle /on togglePowerOn
-        ::wibble::handle /off togglePowerOff
-        ::wibble::handle /status checkStatus
-        ::wibble::handle / notfound
 
-        # Start a server and enter the event loop if not already there.
-        catch {
-            ::wibble::listen 8080
-        }
-    }
-}
-proc ::wibble::check_auth {state} {
-    set auth [dict getnull $state request query auth]
-    set auth [lindex $auth 1]
+	# Auth
 
-    if {$auth eq ""} {
-        return [unauthorized $state]
-    }
+	proc ::wibble::check_auth {state} {
+		set auth [dict getnull $state request query auth]
+		set auth [lindex $auth 1]
 
-    if {$auth != $::plugins::web_api::settings(webserver_authentication_key)} {
-        return [unauthorized $state]
-    }
+		if {$auth eq ""} {
+			return [unauthorized $state]
+		}
 
-    return true;
-}
+		if {$auth != $::plugins::web_api::settings(webserver_authentication_key)} {
+			return [unauthorized $state]
+		}
 
-proc ::wibble::unauthorized {state} {
-    dict set response status 403
-    dict set state response header content-type "" {application/json charset utf-8}
-    dict set response content "{status: \"unauthorized\"}"
-    sendresponse $response
-    return false;
-}
+		return true;
+	}
 
+	proc ::wibble::unauthorized {state} {
+		dict set response status 403
+		dict set state response header content-type "" {application/json charset utf-8}
+		dict set response content "{status: \"unauthorized\"}"
+		sendresponse $response
+		return false;
+	}
 
-proc ::wibble::togglePowerOn {state} {
-    if { ![check_auth $state] } {
-        return;
-    }
+	# Utilities
 
-    start_idle
+	proc ::wibble::return_200_json {{content "{status: ok}"}} {
```

Comment:
Something seems a bit off with your whitespaces here

Reply:
New code: Single tab indentation, consistent with much of the de1app Tcl code I've worked with.
`start_idle`: 4 spaces

We probably should come up with a standard (my preference including no trailing whitespace and each file ends with a newline with no blank lines)

I can see arguments for or against spaces/tabs and 4/8-column indentation